river/tree/splitter/nominal_splitter_classif.py

```diff
@@ -23,72 +29,61 @@ def update(self, att_val, target_val, sample_weight):
         if att_val is None:
             self._missing_weight_observed += sample_weight
         else:
-            try:
-                self._att_val_dist_per_class[target_val]
-            except KeyError:
-                self._att_val_dist_per_class[target_val] = {att_val: 0.0}
-                self._att_val_dist_per_class = dict(
-                    sorted(self._att_val_dist_per_class.items())
-                )
-            try:
-                self._att_val_dist_per_class[target_val][att_val] += sample_weight
-            except KeyError:
-                self._att_val_dist_per_class[target_val][att_val] = sample_weight
-                self._att_val_dist_per_class[target_val] = dict(
-                    sorted(self._att_val_dist_per_class[target_val].items())
-                )
+            self._att_values.add(att_val)
+            self._att_dist_per_class[target_val][att_val] += sample_weight
```

Comment:
In the current implementation, we have a sorting step that I am not confident is covered with default dictionaries.

Reply:
Since all the access operations are using the keys, I believe we can avoid this extra sorting step. But I'll check it again, just to be really sure :)